src/compiler/checker.ts

```diff
@@ -5104,8 +5119,9 @@ namespace ts {
                         }
                     }
                 }
-                context.enclosingDeclaration = saveEnclosingDeclaration;
+                context.enclosingDeclaration = propertySymbol.valueDeclaration || propertySymbol.declarations?.[0] || saveEnclosingDeclaration;

```

Comment:
If we don't set a specific `enclosingDeclaration` when serializing the name, in cases where we don't pass a contextual one (eg, error reporting), we'd print a name like `[iterator]` instead of `[Symbol.iterator]` (because symbol chain lookup bails entirely if there's no enclosing declaration provided). I think we much prefer the later, so now we actually specify an enclosing declaration for the name whenever we can.

Reply:
What's an example where `propertySymbol.declarations?.[0]` is useful?

Reply:
`valueDeclaration` is unset in two cases - one: when the symbol doesn't have a value meaning (as when it comes from an interface), two: when two value symbols merge and have differing value declarations. In both cases, we'd like a context node for the name here.